src/utils/common.js

```diff
@@ -380,3 +380,25 @@ export const groupAndSortByDate = (
 export const isCaseInsensitiveMatch = (a: string, b: string): boolean => {
   return a.toLowerCase() === b.toLowerCase();
 };
+
+export const makePromiseCancelable = (promise: Promise<any>) => {
+  let isCancelled = false;
+
+  const wrappedPromise = new Promise((resolve, reject) => {
+    const cancelledErrorPayload = { isCanceled: true };
+    promise
+      .then((res) => isCancelled
+        ? reject(cancelledErrorPayload)
+        : resolve(res),
+      )
+      .catch((err) => isCancelled
+        ? reject(cancelledErrorPayload)
+        : reject(err),
+      );
```

Comment:
What about something like:

```js
if (isCancelled) {
  const cancel = () => ({ isCanceled: true });

  promise.then(cancel).catch(cancel);
} else {
  promise.then(resolve).catch(reject);
}
```

Reply:
Not really, thats the point that we want to reject cancel promise and in your suggestion it would wait for promise to be completed.

Reply:
Actually I see your point, this actually would work, updated, thanks 👍

Reply:
I'm sorry, this approrach actually doesn't work so had to restore original one. The reason is that `isCanceled` is not yet updated and original callbacks are then built for promise which leads that promise cannot be "cancelled".

Reply:
Please mark as resolved if everything makes sense.

Reply:
Well, my point is that's really confusing to understand what is going on here. I think if we have an `if (..) { .. } else { .. }` it would be more straightforward even if it results on more lines.